### PR TITLE
node: eth publish immediately

### PR DESCRIPTION
**Pull request by bruce-riley**
Change-Id: I242f02d0ed5bcced5ed366a512c07d5757b300e8

### CODE REVIEW

**Comment by evan-gray** [comment]
the "immediate" publish part of #1660's proposal

**Review by evan-gray — DISMISSED** [review]

File: node/pkg/watchers/evm/watcher.go [hunk 1]
```diff
@@ -314,6 +318,19 @@ func (w *Watcher) Run(ctx context.Context) error {
 				}
 
 				for _, msg := range msgs {
+					if msg.ConsistencyLevel == magicNumberPublishImmediately {
+						logger.Info("re-observed message publication transaction, publishing it immediately",
```
Comment (hendrikhofstadt) [inline node/pkg/watchers/evm/watcher.go:322]: maybe use the same log message as below and add a new field for consistency level?

File: node/pkg/watchers/evm/watcher.go [hunk 2]
```diff
@@ -392,6 +409,23 @@ func (w *Watcher) Run(ctx context.Context) error {
 					ConsistencyLevel: ev.ConsistencyLevel,
 				}
 
+				ethMessagesObserved.WithLabelValues(w.networkName).Inc()
+
+				if message.ConsistencyLevel == magicNumberPublishImmediately {
+					logger.Info("found new message publication transaction, publishing it immediately",
```
Comment (hendrikhofstadt) [inline node/pkg/watchers/evm/watcher.go:415]: same here on the log message

**Review by hendrikhofstadt — APPROVED** [review]
Only reviewed the watcher part; looks good, small logging nit

File: node/pkg/watchers/evm/watcher.go [hunk 3]
```diff
@@ -27,6 +27,10 @@ import (
 	"github.com/wormhole-foundation/wormhole/sdk/vaa"
 )
 
+const (
+	magicNumberPublishImmediately = uint8(200)
```
Comment (jynnantonix) [inline node/pkg/watchers/evm/watcher.go:31]: Can we please put constants like this in the sdk rather than having them spread out in various files?

**Review by hendrikhofstadt — DISMISSED** [review]

File: node/pkg/watchers/evm/watcher.go (hunk 1 above)
Comment (bruce-riley) [reply node/pkg/watchers/evm/watcher.go:332]: I actually like having a different message that I can grep for, so I would like to keep this as is.

File: node/pkg/watchers/evm/watcher.go (hunk 3 above)
Comment (bruce-riley) [reply node/pkg/watchers/evm/watcher.go]: Done.

**Review by evan-gray — APPROVED** [review]

**Comment by tbjump** [comment]
So with this change the eth watcher will still only process finalized blocks, correct? https://github.com/wormhole-foundation/wormhole/blob/2552e31b9b7fd11e5caa0621eb264759c382a84f/node/pkg/watchers/evm/connectors/poller.go#L161

**Review by tbjump — CHANGES_REQUESTED** [review]
So with this change the eth watcher will still only process finalized blocks, correct? https://github.com/wormhole-foundation/wormhole/blob/2552e31b9b7fd11e5caa0621eb264759c382a84f/node/pkg/watchers/evm/connectors/poller.go#L157-L164

**Comment by bruce-riley** [comment]
Correct. This change does not impact finality for eth.

**Review by tbjump — APPROVED** [review]
lgtm

**Pull request merged**